server/api/api.go

```diff
@@ -2,41 +2,72 @@ package api
 
 import (
 	"encoding/binary"
-
-	pb "github.com/pl0q1n/goDHT/client_proto"
+	"log"
 
 	"context"
 	"crypto/sha512"
 	"fmt"
+
+	pbClient "github.com/pl0q1n/goDHT/client_proto"
+	pbNode "github.com/pl0q1n/goDHT/node_proto"
 )
 
-// Node..Brief implimentation of node structure for DHT
+// Node...Brief implimentation of node structure for DHT
 type Node struct {
-	hashTable map[uint64]string
-	start     uint64
-	end       uint64
-	id        uint64
+	HashTable       map[uint64][]byte
+	FingerTable     FingerTable
+	FingerTableConn FingerTableConnection
+	start           uint64
+	end             uint64
+	id              uint64
 }
 
 var GlobalNode *Node = &Node{
-	hashTable: make(map[uint64]string),
-	start:     0,
-	end:       0,
-	id:        0,
+	HashTable: make(map[uint64][]byte),
+	FingerTableConn: FingerTableConnection{
+		UpdateChan: make(chan FingerTable, 1),
+	},
+	FingerTable: FingerTable{
+		PreviousEntry: Entry{
+			Host: "",
+			Hash: 0,
+		},
+		SelfEntry: Entry{
+			Host: "",
+			Hash: 0,
+		},
+	},
 }
 
 func SHAToUint64(hash [64]byte) uint64 {
 	return binary.BigEndian.Uint64(hash[:8])
 }
 
+func (node *Node) GetId() uint64 {
+	return node.FingerTable.SelfEntry.Hash
+}
+
 func (node *Node) SetId(host *string) {
 	hashSum := sha512.Sum512([]byte(*host))
-	node.id = SHAToUint64(hashSum)
+	node.FingerTable.SelfEntry.Host = *host
+	node.FingerTable.SelfEntry.Hash = SHAToUint64(hashSum)
+	log.Printf("Log from SetId. HOST: %S, ID: %D", *host, node.FingerTable.SelfEntry.Hash)
+}
+
+func (node *Node) GetRange(start uint64, end uint64) map[uint64][]byte {
+	rangeMap := make(map[uint64][]byte)
+	for key, value := range node.HashTable {
+		if key >= start && key < end {
+			rangeMap[key] = value
+		}
+	}
+	return rangeMap
 }
 
-func (node *Node) ProcessGet(request *pb.GetRequest) *pb.GetResponse {
-	response := &pb.GetResponse{}
-	value, ok := node.hashTable[request.Key]
+func (node *Node) ProcessGet(request *pbClient.GetRequest) *pbClient.GetResponse {
+	response := &pbClient.GetResponse{}
+	key := SHAToUint64(sha512.Sum512(request.Key))
```

Comment:
I think that `toUint64(Sha512(...))` pattern should be extracted into separate function.